src/asl.ts

```diff
@@ -5431,6 +5445,24 @@ export namespace ASLGraph {
     return undefined;
   }
 
+  export function undefinedCompare(
+    left: ASLGraph.JsonPath,
+    right: ASLGraph.Output,
+    operator: ASL.ValueComparisonOperators
+  ) {
+    if (operator === "==" || operator === "===") {
+      if (ASLGraph.isJsonPath(right)) {
+        return ASL.and(
+          ASL.isNotPresent(left.jsonPath),
+          ASL.isNotPresent(right.jsonPath)
+        );
+      } else if (ASLGraph.isLiteralValue(right) && right.value === undefined) {
+        return ASL.isNotPresent(left.jsonPath);
+      }
+    }
+    return undefined;
```

Comment:
Is this case expected?

Reply:
Yeah, returns undefined when there is no comparison to add. If there are no comparisons at all, `false` is returned at runtime.

* `x === undefined` => returns conditional (`isNotPresent(x)`)
* `x === "a"` => return undefined
* `x === y` => return conditional (`isNotPresent(x) && isNotPresent(y)`) 
* `x < undefined` => return undefined